### ISSUE TITLE
android 2.3 - detecting Internet availability

### ISSUE THREAD

**Issue by croccio** [pwittchen/NetworkEvents#37]
not work on android 2.3. i'm connected to wifi but it fall in WIFI_CONNECTED_HAS_NO_INTERNET

**Comment by pwittchen** [comment]
Please, check if this bug occurs on the newest version of the library - v. **1.0.3**. It was released 2 days ago. I fixed Internet connection check in this version and it shouldn't occur any more.

See download section: https://github.com/pwittchen/NetworkEvents#download 

```groovy
dependencies {
  compile 'com.github.pwittchen:networkevents:1.0.3'
}
```

Library works this way: First, you should receive `WIFI_CONNECTED` status and afterwards, you should receive `WIFI_CONNECTED_HAS_INTERNET` status when WiFi network has an Internet connection or `WIFI_CONNECTED_HAS_NO_INTERNET` when WiFi network does not have an Internet connection.

**Comment by johnwatson5412** [comment]
@croccio @pwittchen It's a android system bug.
http://stackoverflow.com/questions/17638398/androids-httpurlconnection-throws-eofexception-on-head-requests

**Comment by johnwatson5412** [comment]
add this code 
connection.setRequestProperty("Accept-Encoding", "");

**Comment by pwittchen** [comment]
This bug is fixed in PR #41. I couldn't reproduce it after fix.
